addons/mail/static/src/components/thread_view_topbar/thread_view_topbar.xml

```diff
@@ -49,6 +49,9 @@
                     <t t-if="threadViewTopbar.thread and threadViewTopbar.thread === messaging.starred">
                         <button class="o_ThreadViewTopbar_unstarAllButton btn btn-secondary" t-att-disabled="threadViewTopbar.threadView.messages.length === 0" t-on-click="threadViewTopbar.onClickUnstarAll">Unstar all</button>
                     </t>
+                    <t t-if="messaging.isCurrentUserGuest and isDebugMode">
```

Comment:
The "Discuss public view" (the page you land on when you follow the invitation link) can also be visited as an internal user, and in that case you would also want the icon to be displayed if you are in debug mode. What I mean is that the condition should not depend on whether the current user is a guest, but rather on whether the current thread viewer is the Discuss public view. Something like `if threadViewTopbar.threadView.threadViewer.discussPublicView`.